Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -8812,24 +8817,28 @@ def fix_security_zone_no_log_setting(
         return result
 
     @staticmethod
-    def get_all_security_rules_in_container(container: Union[Panorama, Firewall, DeviceGroup, Template, Vsys]
-                                            ) -> List[SecurityRule]:
-        """
-        Gets all the security rule objects from the given VSYS or device group and return them.
-        :param container: The object to search for the rules in, as passed to pan-os-python
-        """
+    def get_all_rules_in_container(container: Union[Panorama, Firewall, DeviceGroup, Template, Vsys],
+                                   object_class: Any):
         firewall_rulebase = Rulebase()
         pre_rulebase = PreRulebase()
         post_rulebase = PostRulebase()
         container.add(pre_rulebase)
         container.add(post_rulebase)
         container.add(firewall_rulebase)
```

Comment:
```suggestion
        if object_class is not (SecuirtyRule, NatRule, ....):
            raise ValueError(f'{object_class} must be of type: ....)

        firewall_rulebase = Rulebase()
        pre_rulebase = PreRulebase()
        post_rulebase = PostRulebase()
        container.add(pre_rulebase)
        container.add(post_rulebase)
        container.add(firewall_rulebase)
```

this is not a must, its just to be on the safe side here.

Reply:
Good idea